src/dependencies.rs

```diff
@@ -199,3 +203,85 @@ fn create_dependencies(
         alert_dao,
     }
 }
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use rusqlite::{params, Row};
+
+
+    #[test]
+    fn test_create_dependencies_with_migration_from_03_to_04(){
+        let database = Arc::new(Database::new(
+        //   Connection::open("./testdb2-copy.sqlite").expect("Problem opening db"),
+            Connection::open_in_memory().expect("Couldn't create database!"),
+        ));
+
+        prep_data_03(database.clone());
+        let pragma_variable_name = "user_version";
+        let db_version = database.core_pragma_query(pragma_variable_name);
+
+        assert_eq!(0, db_version);
+
+        create_dependencies(database.clone(), 3);
+
+        let new_db_version = database.core_pragma_query(pragma_variable_name);
+        assert_eq!(1, new_db_version);
+
+        let columns_6 = core_table_info("tcn", database.clone());
+        assert_eq!(6, columns_6.len());
```

Comment:
This test is very lightweight. But ok, this specific migration is more a proof of concept.